pkgs/development/python-modules/adblock/default.nix

```diff
@@ -0,0 +1,59 @@
+{ rustPlatform
+, fetchFromGitHub
+, pipInstallHook
+, pytestCheckHook
+, maturin
+, pkg-config
+, openssl
+, publicsuffix-list
+, isPy27
+, lib
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "adblock";
+  version = "0.4.0";
+  disabled = isPy27;
+
+  # Pypi only has binary releases
+  src = fetchFromGitHub {
+    owner = "ArniDagur";
+    repo = "python-adblock";
+    rev = version;
+    sha256 = "10d6ks2fyzbizq3kb69q478idj0h86k6ygjb6wl3zq3mf65ma4zg";
+  };
+
+  cargoSha256 = "0di05j942rrm2crpdpp9czhh65fmidyrvdp2n3pipgnagy7nchc0";
+
+  format = "pyproject";
+  nativeBuildInputs = [ pipInstallHook maturin pkg-config ];
+  installCheckInputs = [ pytestCheckHook ];
+  buildInputs = [ openssl ];
+  PSL_PATH = "${publicsuffix-list}/share/publicsuffix/public_suffix_list.dat";
+
+  buildPhase = ''
+    runHook preBuild
+    maturin build --release --manylinux off --strip
+    runHook postBuild
+  '';
+
+  # There are no rust tests
+  doCheck = false;
+  # python tests fail with "make sure your test modules/packages have valid Python names"
+  doInstallCheck = false;
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm644 -t dist target/wheels/*.whl
```

Comment:
Please move this into preInstallPhase.

Reply:
This is actually part of the diff for https://github.com/NixOS/nixpkgs/pull/108271. And there is no such thing as `preInstallPhase`. I could move it to preInstall, but that would make it harder to use overrideAttrs to add a new preInstall. So I think it's good like this. It's also copied from pkgs/development/python-modules/retworkx/default.nix. If you still think this should be different, please comment on https://github.com/NixOS/nixpkgs/pull/108271.